protocol/x/perpetuals/keeper/perpetual.go

```diff
@@ -1277,7 +1278,8 @@ func (k Keeper) ModifyOpenInterest(
 	perpetual.OpenInterest = dtypes.NewIntFromBigInt(bigOpenInterest)
 	k.SetPerpetual(ctx, perpetual)
 
-	// TODO(OTE-247): add indexer update logic for open interest change.
+	updatedOIStore := prefix.NewStore(ctx.TransientStore(k.transientStoreKey), []byte(types.UpdatedOIKey))
+	updatedOIStore.Set(lib.Uint32ToKey(perpetualId), bigOpenInterest.Bytes())
```

Comment:
`big.Int` `.Bytes()` function [ignores sign](https://github.com/golang/go/blob/9f3f4c64dbfd37ef9f7113708a706a8099d72fd9/src/math/big/int.go#L523). While this is okay for OI at the current moment, let's prefer using the `Marshal()` function on `dyptes.SerializableInt`